h2o-py/h2o/estimators/estimator_base.py

```diff
@@ -184,53 +187,43 @@ def train(self, x=None, y=None, training_frame=None, offset_column=None, fold_co
                     if xi not in names:
                         raise H2OValueError("Column %s not in the training frame" % xi)
                     xset.add(xi)
+        x = list(xset)
 
-        parms["x"] = list(xset)
```

Comment:
Yes, in line 214 to construct `ignored_columns` which is then passed along to the server.

Although it does seem there's lots of cruft still in the code... for example `x` never has to be transformed from set to a list